File: UserDomain/GetUsersService/helpers/jwt.php
<?php
use Firebase\JWT\JWT;
use Firebase\JWT\Key;

class JWTAuth {
    private static $secret = "6M5X#D6%7Nh*!pkR3HL7F@Fdx";

    public static function createToken($user) {
        $payload = [
            "sub" => $user['_id'],
            "names" => $user['names'],
            "surnames" => $user['surnames'],
            "email" => $user['email'],
            "iat" => time(),
            "exp" => time() + (30 * 24 * 60 * 60) // 30 días
        ];

        return JWT::encode($payload, self::$secret, 'HS256');
    }
}
?>

File: StoryDomain/CreateStoryService/helpers/jwt.go
package helpers

import (
	"time"
	"github.com/dgrijalva/jwt-go"
	"go.mongodb.org/mongo-driver/bson/primitive"
)

func CreateToken(userID primitive.ObjectID, names string, surnames string, email string) (string, error) {
	claims := jwt.MapClaims{
		"sub":      userID.Hex(),  // ✅ Convert ObjectID to hexadecimal string
		"names":    names,
		"surnames": surnames,
		"email":    email,
		"iat":      time.Now().Unix(),
		"exp":      time.Now().Add(time.Hour * 24 * 30).Unix(),
	}

	token := jwt.NewWithClaims(jwt.SigningMethodHS256, claims)
	return token.SignedString([]byte("6M5X#D6%7Nh*!pkR3HL7F@Fdx"))
}